src/storage/txn/actions/flashback_to_version.rs

```diff
@@ -96,7 +96,16 @@ pub fn flashback_to_version<S: Snapshot>(
             *next_lock_key = Some(key);
             break;
         }
-        txn.unlock_key(key.clone(), lock.is_pessimistic_txn());
+        // To guarantee rollback with start ts of the locks
+        reader.start_ts = lock.ts;
+        rollback_lock(
+            txn,
+            reader,
+            key.clone(),
+            &lock,
+            lock.is_pessimistic_txn(),
+            true,
+        )?;
         rows += 1;
         // If the short value is none and it's a `LockType::Put`, we should delete the
         // corresponding key from `CF_DEFAULT` as well.
```

Comment:
This check has been done by the `rollback_lock`, we can remove it.

Reply:
Okay, so the judgment here is just to add the record of deleted row.

Reply:
Maybe we can add an argument to the `rollback_lock` like `&mut rows` to add it inside to avoid writing the duplicated check code.

Reply:
resolved it by using txn.modifies.len().